runtime/tests/error_backtrace1.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+trace=(\
+"    func1 .*\/error_backtrace1.c:20" \
+"    main .*\/error_backtrace1.c:29" \
+"    func2 .*\/error_backtrace1.c:13" \
+"    func1 .*\/error_backtrace1.c:23" \
+"    main .*\/error_backtrace1.c:29" \
+"    func3 .*\/error_backtrace1.c:7" \
+"    func2 .*\/error_backtrace1.c:16" \
+"    func1 .*\/error_backtrace1.c:23" \
+"    main .*\/error_backtrace1.c:29" \
+)
+
+for t in "${trace[@]}"; do
```

Comment:
Can you explain what this is doing?  Why can't you massage the output with sed and then use diff? That would give a much more informative error message.

Reply:
@jclark, What I am doing here is, check whether the expected line is available in the actual backtrace. We have to check line by line because without optimizations actual backtrace will have more trace lines.

Reply:
I would use `join` for that.

Reply:
https://man7.org/linux/man-pages/man1/join.1.html

Reply:
I would avoid using bash extensions as much as possible, and stick to portable sh constructs, which everybody will be familiar with.

Reply:
@jclark, I am not sure whether I understand this comment correctly. Let's say, in actual and expected traces, function names and filenames are identical but line numbers are different. In this case, join commands run successfully. But In our case, we should get an error.
Can you please explain more about this?

Reply:
You can join on whatever you need, including line number. But perhaps the comm command work better for this case.

Reply:
Let me make sure I have understood the problem properly. We have two files F1 and F2 and we need to test whether F2 is the same as F1 except possibly with some lines added. Is that right?

If so, join/comm aren't quite right because they won't ensure lines are in the right order.  So I think the easiest way is to diff F1 F2 (you can diff stdin by using an argument of `-`), and then grep the result for changes that are deletions. If you find any, then the test fails.

Reply:
> Let me make sure I have understood the problem properly. We have two files F1 and F2 and we need to test whether F2 is the same as F1 except possibly with some lines added. Is that right?
> 
Yes, But in the backtrace, we have the absolute file path with file name, that is why I try to match with a pattern to ignore the path(ex: `main .*\/error_backtrace1.c:29`) 
In diff command we always get a difference because paths are different.

Reply:
So then you pipe the output from the error_backtrace through sed to get rid of the absolute paths before piping that into diff.

Reply:
Changed.